libraries/standard/http/src/http_client.c

```diff
@@ -3,59 +3,75 @@
 
 #include "http_client.h"
 #include "private/http_client_internal.h"
-#include "private/http_client_parse.h"
-#include "http_parser/http_parser.h"
 
 /*-----------------------------------------------------------*/
 
 /**
- * @brief An aggregator that represents the user-provided parameters to the
- * #HTTPClient_ReadHeader API function. This will be used as context parameter
- * for the parsing callbacks used by the API function.
+ * @brief Send HTTP bytes over the transport send interface.
+ *
+ * @param[in] pTransport Transport interface.
+ * @param[in] pBuffer HTTP request data to send.
+ * @param[in] bufferLen HTTP request data length.
+ *
+ * @return #HTTP_SUCCESS if successful. If there was a network error or less
+ * bytes than what were specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
  */
-typedef struct findHeaderContext
-{
-    const uint8_t * pField;
-    size_t fieldLen;
-    const uint8_t ** pValueLoc;
-    size_t * pValueLen;
-    uint8_t fieldFound : 1;
-    uint8_t valueFound : 1;
-} findHeaderContext_t;
-
-/*-----------------------------------------------------------*/
+static HTTPStatus_t sendHttpData( const HTTPTransportInterface_t * pTransport,
+                                  const uint8_t * pBuffer,
+                                  size_t bufferLen );
 
 /**
  * @brief Send the HTTP headers over the transport send interface.
  *
  * @param[in] pTransport Transport interface.
  * @param[in] pRequestHeaders Request headers to send, it includes the buffer
  * and length.
+ * @param[in] reqBodyBufLen The length of the request body to be sent. This is
+ * used to generated a Content-Length header.
+ * @param[in] flags Application provided flags to #HTTPClient_Send.
  *
  * @return #HTTP_SUCCESS if successful. If there was a network error or less
- * bytes than what was specified were sent, then #HTTP_NETWORK_ERROR is returned.
+ * bytes than what were specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
  */
 static HTTPStatus_t sendHttpHeaders( const HTTPTransportInterface_t * pTransport,
-                                     const HTTPRequestHeaders_t * pRequestHeaders );
+                                     const HTTPRequestHeaders_t * pRequestHeaders,
+                                     size_t reqBodyBufLen,
+                                     uint32_t flags );
+
+/**
+ * @brief Send the HTTP Content-Length header line over the transport interface.
+ *
+ * @param[in] pTransport Transport interface.
+ * @param[in] contentLength The Content-Length header value to send.
+ *
+ * @return #HTTP_SUCCESS if successful. If there was a network error or less
+ * bytes than what were specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
+ */
+static HTTPStatus_t sendContentLength( const HTTPTransportInterface_t * pTransport,
```

Comment:
All other `send` function names contain `http` word. Can you either remove `http` from each of them (as it they can be considered redundant, being present in the HTTP library source) or add `http` in this for naming consistency?
Alternatively, each of these could use `request` instead of `http` which could be more readable names (imo) ( `sendRequestHeaders`, `sendRequestBody`, `sendRequestContentLengthHeader` )

Reply:
Sure

Reply:
Adding request is also redundant because we never send the response in the client code.